lib/rubocop/cop/rspec/capybara/specific_matcher.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module Capybara
+        # Checks for there is a more specific matcher offered by Capybara.
+        #
+        # @example
+        #
+        #   # bad
+        #   expect(page).to have_selector('button')
+        #   expect(page).to have_no_selector('button.cls')
+        #   expect(page).to have_css('button')
+        #   expect(page).to have_no_css('a.cls', exact_text: 'foo')
+        #   expect(page).to have_css('table.cls')
+        #   expect(page).to have_css('select')
+        #
+        #   # good
+        #   expect(page).to have_button
+        #   expect(page).to have_no_button(class: 'cls')
+        #   expect(page).to have_button
+        #   expect(page).to have_no_link('foo', class: 'cls')
+        #   expect(page).to have_table(class: 'cls')
+        #   expect(page).to have_select
+        #
+        class SpecificMatcher < Base
+          MSG = 'Use `%<good_matcher>s`.'
```

Comment:
While we don't have hard rules on messages, for alternative matchers we usually use `Prefer %<good_matcher>s over %<bad_matcher>s.`

Reply:
I see how It could be confusing, when only some instances of have_selector will register an offense, and the suggested matchers would be different. We can either keep it as it is now, or have something like 'Use have_link instead of have_selector when checking for links'

Reply:
Take the form `Prefer %<good_matcher>s over %<bad_matcher>s.`.